### PR TITLE
Chapter 3 Static Pages 

### PR DESCRIPTION
**Pull request by aisharina99**
chapter 3

### CODE REVIEW

**Comment by aisharina99** [comment]
ready

**Comment by thaodtd-0665** [comment]
there is no file changed

**Comment by thaodtd-0665** [comment]
commented

**Review by huongnt-2545 — COMMENTED** [review]
No changed file in this pull request
image

**Comment by aisharina99** [comment]
ready

**Review by ghost — COMMENTED** [review]
You have 2 commit please make it 1

**Comment by aisharina99** [comment]
ready

File: app/assets/stylesheets/static_pages.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the StaticPages controller here.
+// They will automatically be included in application.css.
```
Comment (ngocvt-0484) [inline app/assets/stylesheets/static_pages.scss]: delete this file because not use

File: app/views/static_pages/help.html.erb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<h1>Help</h1>
+<p>
+Get help on the Ruby on Rails Tutorial at the
+<a href="http://www.railstutorial.org/help">Rails Tutorial
```
Comment (ngocvt-0484) [inline app/views/static_pages/help.html.erb]: please use link_to instead of < a tag

File: app/views/static_pages/home.html.erb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<h1>Sample App</h1>
+<p>
+This is the home page for the
+<a href="http://www.railstutorial.org/">Ruby on
```
Comment (ngocvt-0484) [inline app/views/static_pages/home.html.erb]: same

File: test/controllers/static_pages_controller_test.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
+require "test_helper"
+
+class StaticPagesControllerTest < ActionDispatch::IntegrationTest
+  test "should get home" do
+    get static_pages_home_url
+    assert_response :success
```
Comment (ngocvt-0484) [inline test/controllers/static_pages_controller_test.rb]: dont need do test, please delete this file

File: app/helpers/static_pages_helper.rb [hunk 5]
```diff
@@ -0,0 +1,2 @@
+module StaticPagesHelper
```
Comment (ngocvt-0484) [inline app/helpers/static_pages_helper.rb]: delete this file because not use

**Comment by aisharina99** [comment]
ready

**Review by huongnt-2545 — COMMENTED** [review]
This pull request has some comments not fix yet
please check

**Comment by aisharina99** [comment]
ready

**Review by huongnt-2545 — COMMENTED** [review]
this pull request has 2 commits, please combine in 1
btw, still some comments haven't fix yet

**Comment by aisharina99** [comment]
ready

File: app/views/static_pages/help.html.erb [hunk 6]
```diff
@@ -0,0 +1,10 @@
+<h1>Help</h1>
+<p>
+Get help on the Ruby on Rails Tutorial at the
+link_to href="http://www.railstutorial.org/help">Rails Tutorial
```
Comment (ngocvt-0484) [inline app/views/static_pages/help.html.erb]: wrong syntax
you can refer at https://apidock.com/rails/ActionView/Helpers/UrlHelper/link_to

File: app/views/static_pages/help.html.erb [hunk 7]
```diff
@@ -0,0 +1,10 @@
...
+help page</a>.
+To get help on this sample app, see the
+link_to href="http://www.railstutorial.org/book"><em>Ruby on Rails
```
Comment (ngocvt-0484) [inline app/views/static_pages/help.html.erb]: same
fix it please

**Comment by aisharina99** [comment]
ready

File: app/views/static_pages/help.html.erb (hunk 6 above)
Comment (ghost) [reply app/views/static_pages/help.html.erb]: please fix your cmt

**Review by ghost — COMMENTED** [review]
cmt

**Comment by aisharina99** [comment]
ready

**Review by ghost — COMMENTED** [review]
please fix link_to

**Comment by aisharina99** [comment]
ready

File: app/views/layouts/application.html.erb [hunk 8]
```diff
@@ -1,16 +1,12 @@
...
-
-    <%= stylesheet_link_tag 'application', media: 'all', 'data-turbolinks-track': 'reload' %>
-    <%= javascript_pack_tag 'application', 'data-turbolinks-track': 'reload' %>
-  </head>
-
-  <body>
-    <%= yield %>
-  </body>
+<head>
+<title><%= yield(:title) %> | Ruby on Rails Tutorial Sample App</title>
```
Comment (ghost) [inline app/views/layouts/application.html.erb]: please folow this [pull request](https://github.com/daniadore/sample_app/pull/2/files#diff-f43fe075643e681b2c01c2f853bb0c4299d135b47fcbd4da96890d521c49e3ebL1) fix your code like the pull request please

**Review by ghost — COMMENTED** [review]
cmt

**Comment by aisharina99** [comment]
ready

**Comment by aisharina99** [comment]
ready

File: .gitignore [hunk 9]
```diff
@@ -39,3 +39,8 @@
 yarn-debug.log*
 .yarn-integrity
 /config/database.yml
+.
```
Comment (huongnt-2545) [inline .gitignore]: delete line 42 -> 44

File: app/helpers/application_helper.rb [hunk 10]
```diff
@@ -1,2 +1,8 @@
 module ApplicationHelper
 end
+module ApplicationHelper
+  def full_title page_title = ""
+      base_title = "Ruby on Rails Tutorial Sample App"
+  page_title.empty? ? base_title : "#{page_title} | #{base_title}"
```
Comment (huongnt-2545) [inline app/helpers/application_helper.rb:6]: indent

File: app/helpers/application_helper.rb [hunk 11]
```diff
@@ -1,2 +1,8 @@
 module ApplicationHelper
 end
+module ApplicationHelper
+  def full_title page_title = ""
+      base_title = "Ruby on Rails Tutorial Sample App"
```
Comment (huongnt-2545) [inline app/helpers/application_helper.rb]: indent = 2 spaces
```suggestion
    base_title = "Ruby on Rails Tutorial Sample App"
```

File: app/views/layouts/application.html.erb [hunk 12]
```diff
@@ -1,16 +1,25 @@
...
-    <title>SampleApp</title>
-    <meta name="viewport" content="width=device-width,initial-scale=1">
-    <%= csrf_meta_tags %>
-    <%= csp_meta_tag %>
+ <head>
+   <title>SampleApp</title>
+   <meta name="viewport" content="width=device-width,initial-scale=1">
+   <%= csrf_meta_tags %>
+   <%= csp_meta_tags %>
+...
```
Comment (huongnt-2545) [inline app/views/layouts/application.html.erb]: delete this line

File: app/views/layouts/application.html.erb [hunk 13]
```diff
@@ -1,16 +1,25 @@
...
+   <%= stylesheet_link_tag 'application', media: 'all', 'data-turbolinks-track': 'reload' %>
+   <%= javascript_pack_tag 'application', 'data-turbolinks-track': 'reload' %>
+</head>
 
-    <%= stylesheet_link_tag 'application', media: 'all', 'data-turbolinks-track': 'reload' %>
+ <%= stylesheet_link_tag 'application', media: 'all', 'data-turbolinks-track': 'reload' %>
```
Comment (huongnt-2545) [inline app/views/layouts/application.html.erb]: delete lines from 13 -> 15, these lines are duplicated

File: app/views/layouts/application.html.erb [hunk 14]
```diff
@@ -1,16 +1,25 @@
...
     <%= javascript_pack_tag 'application', 'data-turbolinks-track': 'reload' %>
   </head>
 
   <body>
     <%= yield %>
   </body>
 </html>
+<body>
```
Comment (huongnt-2545) [inline app/views/layouts/application.html.erb]: delete lines from line 21 -> 25, these lines are redundant

File: app/views/static_pages/home.html.erb [hunk 15]
```diff
@@ -0,0 +1,24 @@
+<h1>Sample App</h1>
+<p>
+This is the home page for the
```
Comment (huongnt-2545) [inline app/views/static_pages/home.html.erb]: Please fix all indent in this file

**Review by huongnt-2545 — COMMENTED** [review]
This pull request has some comments not fix yet
please apply it

**Comment by aisharina99** [comment]
ready

File: app/helpers/application_helper.rb (hunk 10 above)
Comment (ngocvt-0484) [reply app/helpers/application_helper.rb:6]: wrong indent

File: app/views/layouts/application.html.erb [hunk 16]
```diff
@@ -6,11 +6,17 @@
     <%= csrf_meta_tags %>
     <%= csp_meta_tag %>
 
+   <%= stylesheet_link_tag 'application', media: 'all', 'data-turbolinks-track': 'reload' %>
+   <%= javascript_pack_tag 'application', 'data-turbolinks-track': 'reload' %>
+</head>
+
     <%= stylesheet_link_tag 'application', media: 'all', 'data-turbolinks-track': 'reload' %>
+<%= stylesheet_link_tag 'application', media: 'all', 'data-turbolinks-track': 'reload' %>
```
Comment (ngocvt-0484) [inline app/views/layouts/application.html.erb:12]: delete it because you defined at line 9

File: app/views/layouts/application.html.erb [hunk 17]
```diff
@@ -6,11 +6,17 @@
...
     <%= javascript_pack_tag 'application', 'data-turbolinks-track': 'reload' %>
   </head>
 
   <body>
     <%= yield %>
   </body>
-</html>
+
+<title><%= yield(:title)%> |Ruby on Rails Tutorial Sample App</title>
```
Comment (ngocvt-0484) [inline app/views/layouts/application.html.erb:20]: ? why not follow the instructions

File: app/views/static_pages/help.html.erb [hunk 18]
```diff
@@ -0,0 +1,30 @@
+<h1>Help</h1>
+<p>
+#Get help on the Ruby on Rails Tutorial at the
+link_to
+=> <a href="http://www.railstutorial.org/help">Rails Tutorial help page</a>.
```
Comment (ngocvt-0484) [inline app/views/static_pages/help.html.erb:5]: use link_to instead of a tag

**Comment by aisharina99** [comment]
ready

**Review by ghost — COMMENTED** [review]
please fix all your comment

**Comment by aisharina99** [comment]
ready

File: app/views/static_pages/help.html.erb (hunk 18 above)
Comment (aisharina99) [reply app/views/static_pages/help.html.erb:5]: why my file did not updated from sublime?

File: app/helpers/application_helper.rb (hunk 10 above)
Comment (aisharina99) [reply app/helpers/application_helper.rb:6]: I have fixed it in sublime but did not come out here?

File: app/views/static_pages/help.html.erb (hunk 18 above)
Comment (aisharina99) [reply app/views/static_pages/help.html.erb:5]: ready

File: app/views/layouts/application.html.erb (hunk 17 above)
Comment (aisharina99) [reply app/views/layouts/application.html.erb:20]: ready

File: app/views/layouts/application.html.erb (hunk 16 above)
Comment (aisharina99) [reply app/views/layouts/application.html.erb:12]: ready

File: app/views/static_pages/home.html.erb (hunk 15 above)
Comment (aisharina99) [reply app/views/static_pages/home.html.erb]: ready

File: app/views/layouts/application.html.erb (hunk 14 above)
Comment (aisharina99) [reply app/views/layouts/application.html.erb]: ready

File: app/views/layouts/application.html.erb (hunk 13 above)
Comment (aisharina99) [reply app/views/layouts/application.html.erb]: ready

File: app/views/layouts/application.html.erb (hunk 12 above)
Comment (aisharina99) [reply app/views/layouts/application.html.erb]: ready

**Review by huongnt-2545 — COMMENTED** [review]
All comments are not fixed yet
You seem follow Rails tutorial not properly, please take carefully step by steps

**Pull request closed without merging**